// @bcodes/ngx-routerlink-delay/fesm2015/bcodes-ngx-routerlink-delay.js
import { Directive, Input, HostListener, ContentChildren, ElementRef, Renderer2, ChangeDetectorRef, NgModule } from '@angular/core';
import { RouterLinkWithHref, Router, ActivatedRoute, RouterLinkActive } from '@angular/router';
import { LocationStrategy, CommonModule } from '@angular/common';
import { timer } from 'rxjs';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkWithHref}
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link.ts
 */
class RouterLinkWithHrefDelay extends RouterLinkWithHref {
    /**
     * @param {?} router
     * @param {?} route
     * @param {?} locationStrategy
     */
    constructor(router, route, locationStrategy) {
        super(router, route, locationStrategy);
        this.navigationDelay = 0;
    }
    /**
     * @param {?} commands
     * @return {?}
     */
    set bcRouterLink(commands) {
        this.routerLink = commands;
    }
    /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    onClick(button, ctrlKey, metaKey, shiftKey) {
        // clone the checks being made in super()
        if (button !== 0 || ctrlKey || metaKey || shiftKey) {
            return true;
        }
        if (typeof this.target === 'string' && this.target !== '_self') {
            return true;
        }
        // Omits Observable.timer 'period' argument so  runs once
        this.timerSubscription = timer(this.navigationDelay)
            .subscribe(t => {
            this.timerSubscription.unsubscribe();
            super.onClick(button, ctrlKey, metaKey, shiftKey);
        });
        return false;
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        super.ngOnDestroy();
        // If the component is destroyed before the timer completes
        if (this.timerSubscription) {
            this.timerSubscription.unsubscribe();
        }
    }
}
RouterLinkWithHrefDelay.decorators = [
    { type: Directive, args: [{
                selector: 'a[bcRouterLink]'
            },] },
];
/** @nocollapse */
RouterLinkWithHrefDelay.ctorParameters = () => [
    { type: Router, },
    { type: ActivatedRoute, },
    { type: LocationStrategy, },
];
RouterLinkWithHrefDelay.propDecorators = {
    "navigationDelay": [{ type: Input },],
    "bcRouterLink": [{ type: Input },],
    "onClick": [{ type: HostListener, args: ['click', ['$event.button', '$event.ctrlKey', '$event.metaKey', '$event.shiftKey'],] },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkActive }
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link_active.ts
 */
class RouterLinkActiveDelay extends RouterLinkActive {
    /**
     * @param {?} router
     * @param {?} element
     * @param {?} renderer
     * @param {?} cdr
     */
    constructor(router, element, renderer, cdr) {
        super(router, element, renderer, cdr);
    }
    /**
     * @param {?} data
     * @return {?}
     */
    set bcRouterLinkActive(data) {
        this.routerLinkActive = data;
    }
}
RouterLinkActiveDelay.decorators = [
    { type: Directive, args: [{
                selector: 'a[bcRouterLinkActive]'
            },] },
];
/** @nocollapse */
RouterLinkActiveDelay.ctorParameters = () => [
    { type: Router, },
    { type: ElementRef, },
    { type: Renderer2, },
    { type: ChangeDetectorRef, },
];
RouterLinkActiveDelay.propDecorators = {
    "linksWithHrefs": [{ type: ContentChildren, args: [RouterLinkWithHrefDelay, { descendants: true },] },],
    "bcRouterLinkActive": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class RouterLinkDelayModule {
}
RouterLinkDelayModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ],
                exports: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ]
            },] },
];
/** @nocollapse */
RouterLinkDelayModule.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { RouterLinkWithHrefDelay, RouterLinkActiveDelay, RouterLinkDelayModule };
