// @bcodes/ngx-routerlink-delay/fesm5/bcodes-ngx-routerlink-delay.js
import { __extends } from 'tslib';
import { Directive, Input, HostListener, ContentChildren, ElementRef, Renderer2, ChangeDetectorRef, NgModule } from '@angular/core';
import { RouterLinkWithHref, Router, ActivatedRoute, RouterLinkActive } from '@angular/router';
import { LocationStrategy, CommonModule } from '@angular/common';
import { timer } from 'rxjs';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkWithHref}
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link.ts
 */
var RouterLinkWithHrefDelay = /** @class */ (function (_super) {
    __extends(RouterLinkWithHrefDelay, _super);
    function RouterLinkWithHrefDelay(router, route, locationStrategy) {
        var _this = _super.call(this, router, route, locationStrategy) || this;
        _this.navigationDelay = 0;
        return _this;
    }
    Object.defineProperty(RouterLinkWithHrefDelay.prototype, "bcRouterLink", {
        set: /**
         * @param {?} commands
         * @return {?}
         */
        function (commands) {
            this.routerLink = commands;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    RouterLinkWithHrefDelay.prototype.onClick = /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    function (button, ctrlKey, metaKey, shiftKey) {
        var _this = this;
        // clone the checks being made in super()
        if (button !== 0 || ctrlKey || metaKey || shiftKey) {
            return true;
        }
        if (typeof this.target === 'string' && this.target !== '_self') {
            return true;
        }
        // Omits Observable.timer 'period' argument so  runs once
        this.timerSubscription = timer(this.navigationDelay)
            .subscribe(function (t) {
            _this.timerSubscription.unsubscribe();
            _super.prototype.onClick.call(_this, button, ctrlKey, metaKey, shiftKey);
        });
        return false;
    };
    /**
     * @return {?}
     */
    RouterLinkWithHrefDelay.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        _super.prototype.ngOnDestroy.call(this);
        // If the component is destroyed before the timer completes
        if (this.timerSubscription) {
            this.timerSubscription.unsubscribe();
        }
    };
    RouterLinkWithHrefDelay.decorators = [
        { type: Directive, args: [{
                    selector: 'a[bcRouterLink]'
                },] },
    ];
    /** @nocollapse */
    RouterLinkWithHrefDelay.ctorParameters = function () { return [
        { type: Router, },
        { type: ActivatedRoute, },
        { type: LocationStrategy, },
    ]; };
    RouterLinkWithHrefDelay.propDecorators = {
        "navigationDelay": [{ type: Input },],
        "bcRouterLink": [{ type: Input },],
        "onClick": [{ type: HostListener, args: ['click', ['$event.button', '$event.ctrlKey', '$event.metaKey', '$event.shiftKey'],] },],
    };
    return RouterLinkWithHrefDelay;
}(RouterLinkWithHref));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkActive }
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link_active.ts
 */
var RouterLinkActiveDelay = /** @class */ (function (_super) {
    __extends(RouterLinkActiveDelay, _super);
    function RouterLinkActiveDelay(router, element, renderer, cdr) {
        return _super.call(this, router, element, renderer, cdr) || this;
    }
    Object.defineProperty(RouterLinkActiveDelay.prototype, "bcRouterLinkActive", {
        set: /**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            this.routerLinkActive = data;
        },
        enumerable: true,
        configurable: true
    });
    RouterLinkActiveDelay.decorators = [
        { type: Directive, args: [{
                    selector: 'a[bcRouterLinkActive]'
                },] },
    ];
    /** @nocollapse */
    RouterLinkActiveDelay.ctorParameters = function () { return [
        { type: Router, },
        { type: ElementRef, },
        { type: Renderer2, },
        { type: ChangeDetectorRef, },
    ]; };
    RouterLinkActiveDelay.propDecorators = {
        "linksWithHrefs": [{ type: ContentChildren, args: [RouterLinkWithHrefDelay, { descendants: true },] },],
        "bcRouterLinkActive": [{ type: Input },],
    };
    return RouterLinkActiveDelay;
}(RouterLinkActive));

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var RouterLinkDelayModule = /** @class */ (function () {
    function RouterLinkDelayModule() {
    }
    RouterLinkDelayModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        RouterLinkWithHrefDelay,
                        RouterLinkActiveDelay
                    ],
                    exports: [
                        RouterLinkWithHrefDelay,
                        RouterLinkActiveDelay
                    ]
                },] },
    ];
    /** @nocollapse */
    RouterLinkDelayModule.ctorParameters = function () { return []; };
    return RouterLinkDelayModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { RouterLinkWithHrefDelay, RouterLinkActiveDelay, RouterLinkDelayModule };
