// @bcodes/ngx-routerlink-delay/esm5/directives/router-link-active-delay.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { RouterLinkActive, Router } from '@angular/router';
import { Directive, ContentChildren, QueryList, ElementRef, Renderer2, ChangeDetectorRef, Input } from '@angular/core';
import { RouterLinkWithHrefDelay } from './router-link-delay';
/**
 * Extends {\@link RouterLinkActive }
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link_active.ts
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
var RouterLinkActiveDelay = /** @class */ (function (_super) {
    tslib_1.__extends(RouterLinkActiveDelay, _super);
    function RouterLinkActiveDelay(router, element, renderer, cdr) {
        return _super.call(this, router, element, renderer, cdr) || this;
    }
    Object.defineProperty(RouterLinkActiveDelay.prototype, "bcRouterLinkActive", {
        set: /**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            this.routerLinkActive = data;
        },
        enumerable: true,
        configurable: true
    });
    /** @nocollapse */
    RouterLinkActiveDelay.ctorParameters = function () { return [
        { type: Router, },
        { type: ElementRef, },
        { type: Renderer2, },
        { type: ChangeDetectorRef, },
    ]; };
    RouterLinkActiveDelay.propDecorators = {
        "linksWithHrefs": [{ type: ContentChildren, args: [RouterLinkWithHrefDelay, { descendants: true },] },],
        "bcRouterLinkActive": [{ type: Input },],
    };
RouterLinkActiveDelay.ɵfac = function RouterLinkActiveDelay_Factory(t) { return new (t || RouterLinkActiveDelay)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Router), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
RouterLinkActiveDelay.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RouterLinkActiveDelay, selectors: [["a", "bcRouterLinkActive", ""]], contentQueries: function RouterLinkActiveDelay_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, RouterLinkWithHrefDelay, 1);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.linksWithHrefs = _t);
    } }, inputs: { bcRouterLinkActive: "bcRouterLinkActive" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RouterLinkActiveDelay, [{
        type: Directive,
        args: [{
                selector: 'a[bcRouterLinkActive]'
            }]
    }], function () { return [{ type: ɵngcc1.Router }, { type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ChangeDetectorRef }]; }, { bcRouterLinkActive: [{
            type: Input
        }], linksWithHrefs: [{
            type: ContentChildren,
            args: [RouterLinkWithHrefDelay, { descendants: true }]
        }] }); })();
    return RouterLinkActiveDelay;
}(RouterLinkActive));
export { RouterLinkActiveDelay };
function RouterLinkActiveDelay_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    RouterLinkActiveDelay.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    RouterLinkActiveDelay.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    RouterLinkActiveDelay.propDecorators;
    /** @type {?} */
    RouterLinkActiveDelay.prototype.linksWithHrefs;
}
