// @bcodes/ngx-routerlink-delay/esm5/directives/router-link-delay.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Directive, Input, HostListener } from '@angular/core';
import { RouterLinkWithHref, Router, ActivatedRoute } from '@angular/router';
import { LocationStrategy } from '@angular/common';
import { timer } from 'rxjs';
/**
 * Extends {\@link RouterLinkWithHref}
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link.ts
 */
var RouterLinkWithHrefDelay = /** @class */ (function (_super) {
    tslib_1.__extends(RouterLinkWithHrefDelay, _super);
    function RouterLinkWithHrefDelay(router, route, locationStrategy) {
        var _this = _super.call(this, router, route, locationStrategy) || this;
        _this.navigationDelay = 0;
        return _this;
    }
    Object.defineProperty(RouterLinkWithHrefDelay.prototype, "bcRouterLink", {
        set: /**
         * @param {?} commands
         * @return {?}
         */
        function (commands) {
            this.routerLink = commands;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    RouterLinkWithHrefDelay.prototype.onClick = /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    function (button, ctrlKey, metaKey, shiftKey) {
        var _this = this;
        // clone the checks being made in super()
        if (button !== 0 || ctrlKey || metaKey || shiftKey) {
            return true;
        }
        if (typeof this.target === 'string' && this.target !== '_self') {
            return true;
        }
        // Omits Observable.timer 'period' argument so  runs once
        this.timerSubscription = timer(this.navigationDelay)
            .subscribe(function (t) {
            _this.timerSubscription.unsubscribe();
            _super.prototype.onClick.call(_this, button, ctrlKey, metaKey, shiftKey);
        });
        return false;
    };
    /**
     * @return {?}
     */
    RouterLinkWithHrefDelay.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        _super.prototype.ngOnDestroy.call(this);
        // If the component is destroyed before the timer completes
        if (this.timerSubscription) {
            this.timerSubscription.unsubscribe();
        }
    };
    RouterLinkWithHrefDelay.decorators = [
        { type: Directive, args: [{
                    selector: 'a[bcRouterLink]'
                },] },
    ];
    /** @nocollapse */
    RouterLinkWithHrefDelay.ctorParameters = function () { return [
        { type: Router, },
        { type: ActivatedRoute, },
        { type: LocationStrategy, },
    ]; };
    RouterLinkWithHrefDelay.propDecorators = {
        "navigationDelay": [{ type: Input },],
        "bcRouterLink": [{ type: Input },],
        "onClick": [{ type: HostListener, args: ['click', ['$event.button', '$event.ctrlKey', '$event.metaKey', '$event.shiftKey'],] },],
    };
    return RouterLinkWithHrefDelay;
}(RouterLinkWithHref));
export { RouterLinkWithHrefDelay };
function RouterLinkWithHrefDelay_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    RouterLinkWithHrefDelay.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    RouterLinkWithHrefDelay.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    RouterLinkWithHrefDelay.propDecorators;
    /** @type {?} */
    RouterLinkWithHrefDelay.prototype.navigationDelay;
    /** @type {?} */
    RouterLinkWithHrefDelay.prototype.timerSubscription;
}
