// @bcodes/ngx-routerlink-delay/__ivy_ngcc__/fesm2015/bcodes-ngx-routerlink-delay.js
import { Directive, Input, HostListener, ContentChildren, ElementRef, Renderer2, ChangeDetectorRef, NgModule } from '@angular/core';
import { RouterLinkWithHref, Router, ActivatedRoute, RouterLinkActive } from '@angular/router';
import { LocationStrategy, CommonModule } from '@angular/common';
import { timer } from 'rxjs';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkWithHref}
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link.ts
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
import * as ɵngcc2 from '@angular/common';
class RouterLinkWithHrefDelay extends RouterLinkWithHref {
    /**
     * @param {?} router
     * @param {?} route
     * @param {?} locationStrategy
     */
    constructor(router, route, locationStrategy) {
        super(router, route, locationStrategy);
        this.navigationDelay = 0;
    }
    /**
     * @param {?} commands
     * @return {?}
     */
    set bcRouterLink(commands) {
        this.routerLink = commands;
    }
    /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    onClick(button, ctrlKey, metaKey, shiftKey) {
        // clone the checks being made in super()
        if (button !== 0 || ctrlKey || metaKey || shiftKey) {
            return true;
        }
        if (typeof this.target === 'string' && this.target !== '_self') {
            return true;
        }
        // Omits Observable.timer 'period' argument so  runs once
        this.timerSubscription = timer(this.navigationDelay)
            .subscribe(t => {
            this.timerSubscription.unsubscribe();
            super.onClick(button, ctrlKey, metaKey, shiftKey);
        });
        return false;
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        super.ngOnDestroy();
        // If the component is destroyed before the timer completes
        if (this.timerSubscription) {
            this.timerSubscription.unsubscribe();
        }
    }
}
RouterLinkWithHrefDelay.ɵfac = function RouterLinkWithHrefDelay_Factory(t) { return new (t || RouterLinkWithHrefDelay)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Router), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ActivatedRoute), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.LocationStrategy)); };
RouterLinkWithHrefDelay.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RouterLinkWithHrefDelay, selectors: [["a", "bcRouterLink", ""]], hostBindings: function RouterLinkWithHrefDelay_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function RouterLinkWithHrefDelay_click_HostBindingHandler($event) { return ctx.onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey); });
    } }, inputs: { navigationDelay: "navigationDelay", bcRouterLink: "bcRouterLink" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
/** @nocollapse */
RouterLinkWithHrefDelay.ctorParameters = () => [
    { type: Router, },
    { type: ActivatedRoute, },
    { type: LocationStrategy, },
];
RouterLinkWithHrefDelay.propDecorators = {
    "navigationDelay": [{ type: Input },],
    "bcRouterLink": [{ type: Input },],
    "onClick": [{ type: HostListener, args: ['click', ['$event.button', '$event.ctrlKey', '$event.metaKey', '$event.shiftKey'],] },],
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RouterLinkWithHrefDelay, [{
        type: Directive,
        args: [{
                selector: 'a[bcRouterLink]'
            }]
    }], function () { return [{ type: ɵngcc1.Router }, { type: ɵngcc1.ActivatedRoute }, { type: ɵngcc2.LocationStrategy }]; }, { navigationDelay: [{
            type: Input
        }], bcRouterLink: [{
            type: Input
        }], 
    /**
     * @param {?} button
     * @param {?} ctrlKey
     * @param {?} metaKey
     * @param {?} shiftKey
     * @return {?}
     */
    onClick: [{
            type: HostListener,
            args: ['click', ['$event.button', '$event.ctrlKey', '$event.metaKey', '$event.shiftKey']]
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Extends {\@link RouterLinkActive }
 * @see https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link_active.ts
 */
class RouterLinkActiveDelay extends RouterLinkActive {
    /**
     * @param {?} router
     * @param {?} element
     * @param {?} renderer
     * @param {?} cdr
     */
    constructor(router, element, renderer, cdr) {
        super(router, element, renderer, cdr);
    }
    /**
     * @param {?} data
     * @return {?}
     */
    set bcRouterLinkActive(data) {
        this.routerLinkActive = data;
    }
}
RouterLinkActiveDelay.ɵfac = function RouterLinkActiveDelay_Factory(t) { return new (t || RouterLinkActiveDelay)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Router), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
RouterLinkActiveDelay.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RouterLinkActiveDelay, selectors: [["a", "bcRouterLinkActive", ""]], contentQueries: function RouterLinkActiveDelay_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, RouterLinkWithHrefDelay, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.linksWithHrefs = _t);
    } }, inputs: { bcRouterLinkActive: "bcRouterLinkActive" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
/** @nocollapse */
RouterLinkActiveDelay.ctorParameters = () => [
    { type: Router, },
    { type: ElementRef, },
    { type: Renderer2, },
    { type: ChangeDetectorRef, },
];
RouterLinkActiveDelay.propDecorators = {
    "linksWithHrefs": [{ type: ContentChildren, args: [RouterLinkWithHrefDelay, { descendants: true },] },],
    "bcRouterLinkActive": [{ type: Input },],
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RouterLinkActiveDelay, [{
        type: Directive,
        args: [{
                selector: 'a[bcRouterLinkActive]'
            }]
    }], function () { return [{ type: ɵngcc1.Router }, { type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ChangeDetectorRef }]; }, { bcRouterLinkActive: [{
            type: Input
        }], linksWithHrefs: [{
            type: ContentChildren,
            args: [RouterLinkWithHrefDelay, { descendants: true }]
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class RouterLinkDelayModule {
}
RouterLinkDelayModule.ɵfac = function RouterLinkDelayModule_Factory(t) { return new (t || RouterLinkDelayModule)(); };
RouterLinkDelayModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: RouterLinkDelayModule });
RouterLinkDelayModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
/** @nocollapse */
RouterLinkDelayModule.ctorParameters = () => [];
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(RouterLinkDelayModule, { declarations: function () { return [RouterLinkWithHrefDelay, RouterLinkActiveDelay]; }, imports: function () { return [CommonModule]; }, exports: function () { return [RouterLinkWithHrefDelay, RouterLinkActiveDelay]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RouterLinkDelayModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ],
                exports: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { RouterLinkWithHrefDelay, RouterLinkActiveDelay, RouterLinkDelayModule };
