// @bcodes/ngx-routerlink-delay/esm5/router-link-delay.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterLinkWithHrefDelay } from './directives/router-link-delay';
import { RouterLinkActiveDelay } from './directives/router-link-active-delay';
import * as ɵngcc0 from '@angular/core';
var RouterLinkDelayModule = /** @class */ (function () {
    function RouterLinkDelayModule() {
    }
    /** @nocollapse */
    RouterLinkDelayModule.ctorParameters = function () { return []; };
RouterLinkDelayModule.ɵfac = function RouterLinkDelayModule_Factory(t) { return new (t || RouterLinkDelayModule)(); };
RouterLinkDelayModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: RouterLinkDelayModule });
RouterLinkDelayModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(RouterLinkDelayModule, { declarations: function () { return [RouterLinkWithHrefDelay,
        RouterLinkActiveDelay]; }, imports: function () { return [CommonModule]; }, exports: function () { return [RouterLinkWithHrefDelay,
        RouterLinkActiveDelay]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RouterLinkDelayModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ],
                exports: [
                    RouterLinkWithHrefDelay,
                    RouterLinkActiveDelay
                ]
            }]
    }], function () { return []; }, null); })();
    return RouterLinkDelayModule;
}());
export { RouterLinkDelayModule };
function RouterLinkDelayModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    RouterLinkDelayModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    RouterLinkDelayModule.ctorParameters;
}
